core/src/block.rs

```diff
@@ -474,8 +474,11 @@ mod valid {
 
         /// Add additional signatures for [`Self`].
         #[must_use]
-        pub fn sign(self, key_pair: &KeyPair) -> ValidBlock {
-            ValidBlock(self.0.sign(key_pair))
+        pub fn sign(self, key_pair: &KeyPair, topology: &Topology) -> ValidBlock {
+            let node_pos = topology
+                .position(key_pair.public_key())
+                .expect("BUG: Node is not in topology");
```

Comment:
Why you decided to panic here? Maybe allow downstream users to handle error condition?

Reply:
I was thinking that it indicates a bug in consensus. The node that is signing must be present in the topology. This must never happen, I consider it the same as accessing array out of bounds.

I will reconsider

Reply:
encountering this panic means that `Role::Undefined` is trying to sign a block. This mustn't happen in a correct consensus so panic is appropriate